packages/cursorless-vscode/src/CommandHistory.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  ActionDescriptor,
+  CommandComplete,
+  CommandHistoryItem,
+  Disposable,
+  FileSystem,
+  ReadOnlyHatMap,
+} from "@cursorless/common";
+import type {
+  CommandRunner,
+  CommandRunnerDecorator,
+} from "@cursorless/cursorless-engine";
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+import * as vscode from "vscode";
+
+const filePrefix = "cursorlessCommandHistory";
+const settingSection = "cursorless";
+const settingName = "commandHistory";
+const settingFullName = `${settingSection}.${settingName}`;
+
+export class CommandHistory implements CommandRunnerDecorator {
```

Comment:
jsdoc

Reply:
Not quite sure what to write in this case but I added a small comment